contracts/StakeManagerV2.sol

```diff
@@ -0,0 +1,331 @@
+            revert DelegationAmountTooSmall();
+
+        uint256 shares = IStakeCredit(
+            STAKE_HUB.getValidatorCreditContract(_fromOperator)
+        ).getSharesByPooledBNB(_amount);
+
+        STAKE_HUB.redelegate(_fromOperator, _toOperator, shares, true);
+
+        emit Redelegated(_fromOperator, _toOperator, _amount);
+    }
+
+    /// @notice Delegate BNB to the preferred operator without minting BnbX.
+    /// @dev This function is useful for boosting staking rewards and for initial
+    ///      Fusion hardfork migration without affecting the token supply.
+    /// @dev Can only be called by an address with the DEFAULT_ADMIN_ROLE.
+    function delegateWithoutMinting()
+        external
+        payable
+        override
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        if (msg.value < STAKE_HUB.minDelegationBNBChange())
+            revert DelegationAmountTooSmall();
+
+        address preferredOperatorAddress = operatorRegistry
+            .preferredDepositOperator();
+        STAKE_HUB.delegate{value: msg.value}(preferredOperatorAddress, true);
+
+        emit Delegated(preferredOperatorAddress, msg.value);
+    }
+
+    /// @notice Set the operator registry address.
+    /// @param _operatorRegistry The new operator registry address.
+    function setOperatorRegistry(
+        address _operatorRegistry
+    ) external override onlyRole(DEFAULT_ADMIN_ROLE) {
+        if (_operatorRegistry == address(0)) revert ZeroAddress();
+
+        operatorRegistry = IOperatorRegistry(_operatorRegistry);
+        emit OperatorRegistryUpdated(address(operatorRegistry));
+    }
+
+    /// @notice Set the BnbX address.
+    /// @param _bnbX The new BnbX address.
+    function setBnbX(
+        address _bnbX
+    ) external override onlyRole(DEFAULT_ADMIN_ROLE) {
+        if (_bnbX == address(0)) revert ZeroAddress();
+
+        bnbX = IBnbX(_bnbX);
+        emit BnbXUpdated(address(bnbX));
+    }
+
+    /// @notice Pause or unpause the contract.
+    function togglePause() external override onlyRole(DEFAULT_ADMIN_ROLE) {
+        paused() ? _unpause() : _pause();
+    }
+
+    /// @notice Internal function to undelegate BNB from an operator.
+    /// @param _operator The address of the operator.
+    /// @param _amount The amount of BNB to undelegate.
+    /// @return The amount of BNB actually undelegated.
+    function _undelegate(
+        address _operator,
+        uint256 _amount
+    ) internal returns (uint256) {
+        address creditContract = STAKE_HUB.getValidatorCreditContract(
+            _operator
+        );
+        uint256 pooledBnb = IStakeCredit(creditContract).getPooledBNB(
+            address(this)
+        );
+
+        uint256 amount2WithdrawFromOperator = (pooledBnb <= _amount)
+            ? pooledBnb
+            : _amount;
+
+        uint256 shares = IStakeCredit(creditContract).getSharesByPooledBNB(
+            amount2WithdrawFromOperator
+        );
+        STAKE_HUB.undelegate(_operator, shares);
```

Comment:
What happens when shares are 0? Effectively when validator has 0 BNB staked with them.

Reply:
it would revert : [ref](https://github.com/bnb-chain/bsc-genesis-contract/blob/9560688b65b7f8c180a7b70bda985da292391d94/contracts/BC_fusion/StakeCredit.sol#L114)

Reply:
but as per this impl, it will never revert as we check the amount delegated to the operator